gallery_dl/extractor/kemonoparty.py

```diff
@@ -52,7 +52,12 @@ def items(self):
             append = files.append
             file = post["file"]
 
-            if file:
+            # issue 1667 - patreon posts duplicate their main image
+            # in the attachments (sometimes)
+            should_skip_main_file = post["service"] == "patreon" \
+            and not post["attachments"]
+
+            if file and not should_skip_main_file:
                 file["type"] = "file"
                 append(file)
```

Comment:
I'd just do the following to not have to fight the linter:

``` python
            if post["service"] != "patreon" or not post["attachments"]:
                file = post["file"]
                if file:
                    file["type"] = "file"
                    append(file)
```

Or this, to set a file's `type` and not change the general metadata structure, even if said file gets ignored for download:
``` python
            file = post["file"]

            if file:
                file["type"] = "file"
                if post["service"] != "patreon" or not post["attachments"]:
                    append(file)
```

Reply:
No, I'm sorry for being too lazy to install the linter. It's proper now.